partitura/io/matchfile_base.py

```diff
@@ -799,7 +799,20 @@ def prepare_kwargs_from_matchline(
         matchline: str,
         snote_class: BaseSnoteLine,
         version: Version,
-    ) -> Dict:
+        pos: int = 0,
+    ) -> Dict:       
+        # this is very roundabout, but since this is a class method
+        # we can't use instance properties without a dummy instance of cls.
+        # and the note class pattern is only set when instantiated
+        # for some note classes (expecially v0)
+        dummy_snote = snote_class(version = version,anchor= 0,note_name="C",
```

Comment:
Maybe it would be better to create the dummy instances outside of the method (i.e., as global variables)? Otherwise, calling this method will always create new instances of dummy_snote and dummy_instance.

Reply:
I can create the dummy snotes and notes outside but not the dummy instances for the class itself. so some instances are always created, yes

Reply:
Ok, in that case we can leave it as it is. If it is becomes too slow for large datasets (e.g., future extensions to (n)ASAP), maybe we should consider refactoring.